sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/DefaultAggregationSelector.java

```diff
@@ -24,6 +26,33 @@ static DefaultAggregationSelector getDefault() {
     return instrumentType -> Aggregation.defaultAggregation();
   }
 
+  /**
+   * Returns a composed default aggregation selector which uses the given {@code aggregation} for
+   * the given {@code instrumentType}, and defers to this for other instrument types.
+   *
+   * <p>For example, the following produces a selector which drops histograms and uses the default
+   * aggregation for other instruments:
+   *
+   * <pre>{@code
+   * // DefaultAggregationSelector selector =
+   * //   DefaultAggregationSelector.getDefault()
+   * //     .compose(InstrumentType.HISTOGRAM, Aggregation.drop());
+   * }</pre>
+   *
+   * @since 1.16.0
+   */
+  default DefaultAggregationSelector compose(
+      InstrumentType instrumentType, Aggregation aggregation) {
```

Comment:
This isn't needed for this change, but I think it really improves the ergonomics of defining `DefaultAggregationSelector` instances. Inspired by [Function#compose](https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#compose-java.util.function.Function-).

Reply:
Perhaps can be named `withAggregation` or something, this feels more like a "mutation" of the selector rather than composition (usually you compose two of the same type)

Reply:
I like that, but how about just `with`? The class is annotated with a `FunctionalInterface` so its likely to stay close to its current form and not need to disambiguate between with "aggregation" and with "something else".